src/content/docs/browser/browser-monitoring/browser-pro-features/auto-logging.mdx

```diff
@@ -0,0 +1,78 @@
+- `console.info`
+- `console.log` (this will be treated the same as `INFO` level)
+- `console.debug`
+- `console.trace`
+
+Data passed through the console methods may go through deserialization and obfuscation.  Depending on the size and frequency, this may negatively impact application performance as well as data costs.  In general, it is NOT recommended to pass in large objects or large amounts of data into console methods.
+
+By default, browser event data is stored for 8 days, but actual data retention depends on your account.
+
+## Get started [#get-started]
+
+<Steps>
+  <Step>
+    ### Enable auto logging [#enable-configure-settings]
+
+    1. Go to <DNT>**[one.newrelic.com](https://one.newrelic.com/all-capabilities) > All Capabilities > Browser**</DNT>.
+    2. Select your browser app.
+    3. In the left-hand menu, click  <DNT>**Application settings**</DNT>.
+    4. On the Application settings page, make sure <DNT>**Pro**</DNT> or <DNT>**Pro + SPA**</DNT> browser agent is selected. Auto logging is not available for the Lite browser agent.
+    5. Toggle ON <DNT>**Auto logging**</DNT> settings.
+  </Step>
+
+  <Step>
+    ### Configure sampling rates [#configure-sampling-rates]
+
+    Set a sampling rate (0%-100%) for the following samples:
+
+    * <DNT>**User sessions**</DNT> records a random sample of all user sessions.
+
+      For example, if you set the session sampling rate to 50%, it means that:
+
+    * Half of all user sessions will automatically collect log events.
+  </Step>
+
+  <Step>
+    ### View log events [#view-events]
+
+    You can find logging data in two places:
+
+    * On the <DNT>**Logs**</DNT> page:
+    
+      1. Go to: <DNT>**[one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > Logs**</DNT>. For details on what you can do in the UI, see [Logs UI](/docs/logs/ui-data/use-logs-ui).
+
+         You can also query the `Log` data type. Here's a simple example NRQL query:
+
+        ```sql
+        SELECT * FROM Log
+        ```
+
+        You can also use NerdGraph, our GraphQL-format API to [query data](/docs/apis/nerdgraph/examples/nerdgraph-nrql-tutorial), or to [configure log management](/docs/apis/nerdgraph/examples/nerdgraph-log-parsing-rules-tutorial).
+
+    * On the <DNT>**Errors inbox**</DNT> page:
+
+      1. In the left-hand browser menu, click <DNT>**Errors**</DNT>.
+      2. Click on the <DNT>**Triage**</DNT> and <DNT>**Group errors**</DNT> pages to see logs attached to errors.
+  </Step>
+</Steps>
+
+## Data consumption [#data-consumption]
+
+Logs follow the same consumption pricing as your other browser bytes. The amount of bytes produced depends on the count and length of messages.
+
+With the auto logging feature, it is no longer required to invoke the browser APIs `newrelic.log` or `newrelic.wrapLogger` to capture browser logs.  Those methods will continue to work, but it's not advisable to have both as duplicate log messages will incur additional data consumption.
```

Comment:
Should double check this.  I think the browser agent will prevent the methods from being double wrapped.  That said, it will honor the first wrapper (which should be our internal wrapping), meaning the default `level` would be applied instead of any custom `level` set by the customer when double implemented.  It is best to avoid the double wrapping altogether when possible

Reply:
I was mistakenly thinking of the example where we wrapped `loggers` but that doesn't really make a lot of sense.

Double-checked when we have both (auto + explicit wrapping around `console`), the auto logging "wins" and there're no data duplication.  However, I do seem to lose the custom attributes with auto logging.  Is this an expected behavior or a missing behavior?

Reply:
Closing this as we've updated the behavior such that wrappers will be able to update the custom attributes.